File: app/src/main/java/com/maiya/call/phone/utils/ThreadManager.java
package com.maiya.call.phone.utils;

import androidx.annotation.NonNull;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.ThreadFactory;

/**
 * Created by ymc on 2020/11/19.
 *
 * @Description
 */
public class ThreadManager {

    private static ExecutorService executorService = Executors.newCachedThreadPool(new ThreadFactory() {
        @Override
        public Thread newThread(@NonNull Runnable runnable) {
            Thread result = new Thread(runnable, "call_thread");
            result.setDaemon(false);
            return result;
        }
    });

    public static void execute(Runnable runnable) {
        executorService.execute(runnable);
    }

}

File: app/src/main/java/com/maiya/call/phone/utils/ThreadManager.kt
package com.maiya.call.phone.utils

import java.util.concurrent.Executors

/**
 * Created by ymc on 2020/11/19.
 *
 * @Description
 */
object ThreadManager {
    private val executorService = Executors.newCachedThreadPool { runnable ->
        val result = Thread(runnable, "call_thread")
        result.isDaemon = false
        result
    }

    fun execute(runnable: Runnable?) {
        executorService.execute(runnable)
    }
}